core/src/services/hdfs_native/reader.rs

```diff
@@ -15,23 +15,27 @@
 // specific language governing permissions and limitations
 // under the License.
 
+use bytes::Bytes;
 use hdfs_native::file::FileReader;
 
 use crate::raw::*;
+use crate::services::hdfs_native::error::parse_hdfs_error;
 use crate::*;
 
 pub struct HdfsNativeReader {
-    _f: FileReader,
+    f: FileReader,
+    size: usize,
 }
 
 impl HdfsNativeReader {
-    pub fn new(f: FileReader) -> Self {
-        HdfsNativeReader { _f: f }
+    pub fn new(f: FileReader, size: usize) -> Self {
+        HdfsNativeReader { f, size }
     }
 }
 
 impl oio::Read for HdfsNativeReader {
     async fn read(&mut self) -> Result<Buffer> {
-        todo!()
+        let bytes: Bytes = self.f.read(self.size).await.map_err(parse_hdfs_error)?;
```

Comment:
This will read all data in memory at once. Please use `read_range_stream` to get a bytes stream and yield bytes everytime users call `read`.